src/main/scala/mesosphere/marathon/core/task/update/impl/TaskStatusUpdateProcessorImpl.scala

```diff
@@ -47,7 +47,20 @@ class TaskStatusUpdateProcessorImpl @Inject() (
     val taskId = Task.Id(status.getTaskId)
     val taskCondition = TaskCondition(status)
 
+    def taskIsUnknown(instance: Instance, taskId: Task.Id) = {
+      instance.tasksMap.get(taskId).isEmpty
+    }
+
     instanceTracker.instance(taskId.instanceId).flatMap {
+      case Some(instance) if taskIsUnknown(instance, taskId) =>
+        if (killWhenUnknown(taskCondition)) {
+          killUnknownTaskTimer {
+            logger.warn(s"Kill unknown ${taskId} because it's not tracked by marathon")
```

Comment:
I think we could be very explicit here, since this is an exceptional case where we might want to have more info in the logs if it occurs. Something like
> `Kill ${taskId} because it's unknown to marathon. The related instance ${instanceId} is associated with ${instance.tasksMap.keys}`